### DISCUSSION TITLE
jj restore during a rebase conflict?

### DISCUSSION THREAD

**Discussion by sirn** [jj-vcs/jj#4671] [Q&A] [answered]
Hi, I'm currently in this scenario where I have two branches, `next` (a very big rewrite) and `main` (current codebase). I wanted to rebase `next` onto `main`, so I did:

```
$ jj rebase -b next -d main
```

This worked as expected, but there were a lot of conflicts, so I did the normal conflict resolution flow:

```
$ jj new w
Working copy now at: qyxulrwm 69d9c833 (conflict) (empty) (no description set)
Parent commit      : wlyyqklq ce1d214f (conflict) rewrite
```

Since I don't care about changes from `main`, I read in [working-copy.md](https://github.com/martinvonz/jj/blob/35e47e7fa02ac8d41c98e43fa98841fd98833ed3/docs/working-copy.md) that "[y]ou can use `jj restore` to choose one side of the conflict, but there's no way to even see where the involved parts came from." So I tried to `jj restore --from w -- src/` to get `src/` restored to the original state of the `next` branch before rebasing:

```
$ jj restore --from w -- src/
Nothing changed.
```

This, however, doesn't work because `w` is now the commit with conflict markers. I can do `jj restore --from main` to choose the `main` side, but I want to discard the changes from `main`. Is there any way to restore from the pre-conflict marker-`next` branch in this case?

(I cannot directly restore `--from next@remote` either because this is the first commit where more changes are made upon)

Thanks!

**Comment by yuja** [comment, marked as answer]
You can find old state by `jj evolog -r @-` (or `jj op log --op-diff`), and use the commit id (not the change id) to restore the files from.

**Reply by sirn** [reply]
Thank you, that worked perfectly!
